plugin/src/main/kotlin/com/nbadal/ktlint/KtlintFormat.kt

```diff
@@ -54,24 +53,18 @@ internal fun ktlintFormat(
     val canNotBeAutoCorrectedErrors = mutableListOf<LintError>()
     val ignoredErrors = mutableListOf<LintError>()
     try {
-        // If file contains unsaved changed, those will not be picked up by the KtlintRuleEngine.
-        // Add new Code factory methode to create snippet with a virtual path. The ".editorconfig" will be read from
-        // this virtual path.
-        // Given code snippet below which does not contain lint violations:
-        //     fun foo(
-        //         a: String,
-        //         b: String,
-        //         c: Int
-        //     ) = a + b + c
-        // Joining any two consecutive lines with shortcut CTRL-SHIFT-J should result in a new lint violation. But,
-        // no such violation is found as the shortcut does not directly save the changes. Whenever the lines are
-        // merged by deleting the newline with either backspace or delete, does result in merging the lines *and*
-        // saving the changes before invoking the annotator.
         val formattedCode =
             KtLintRuleEngine(
                 editorConfigOverride = EMPTY_EDITOR_CONFIG_OVERRIDE,
                 ruleProviders = ruleProviders,
-            ).format(Code.fromFile(File(filePath))) { error, corrected ->
+            ).format(
+                // The psiFile may contain unsaved changes. So create a snippet based on content of the psiFile *and*
+                // with the same path as that psiFile so that the correct '.editorconfig' is picked up by ktlint.
+                Code.fromSnippet(
+                    content = psiFile.text,
+                    path = psiFile.virtualFile.toNioPath(),
+                )
```

Comment:
This functionality will only be supported with next version (1.1.0) of Ktlint. If you want to run locally with this branch, just comment out line `path = psiFile.virtualFile.toNioPath(),`. This change ensures that changes in the PsiFile which are not yet save to the physical file will be processed by Ktlint. The path is only needed so that Ktlint can locate the correct `.editorconfig` file to read the settings.